drivers/net/xaptum/f_psock_socket.c

```diff
@@ -0,0 +1,328 @@
+/**
+ * @file f_psock_socket.c
+ * @brief Create the psock socket type
+ *  This creates a new socket type for proxying to the connected host device
+ *  This part of the module communicates with the psock_proxy part
+ * @author Jeroen Z
+ */
+
+#include "f_psock_proxy.h"
+
+#include <linux/net.h>
+#include <net/sock.h>
+
+#define PF_XPT PF_NFC
```

Comment:
Since we're integrating into the Linux tree, let's define an actual new packet family, not steal the NFC number. Also, change the name from `PF_XPT` to `PF_PSOCK`.

In [`include/linux/socket.h`](https://github.com/torvalds/linux/blob/1a9df9e29c2afecf6e3089442d429b377279ca3c/include/linux/socket.h#L212)
`#define AF_PSOCK 45`, increment  `AF_MAX` to 46, and `#define PF_PSOCK AF_PSOCK` to the list below.